pkg/sql/sqlbase/structured.go

```diff
@@ -4405,7 +4405,7 @@ func (desc *TypeDescriptor) HydrateTypeInfo(typ *types.T) error {
 // a type and also sets the name in the metadata to the passed in name.
 // This is used when hydrating a type with a known qualified name.
 func (desc *TypeDescriptor) HydrateTypeInfoWithName(typ *types.T, name *tree.TypeName) error {
-	typ.TypeMeta.Name = name
+	typ.TypeMeta.Name = types.MakeUserDefinedTypeName(name.Catalog(), name.Schema(), name.Object())
```

Comment:
I wonder if it would be good to have an `isHydrated` method that we could check from time to time as a sanity check. Could that just check whether `TypeMeta.Name` is nil? Otherwise I fear we'll end up having to debug a lot of confusing issues instead of clear panics/errors that say "Type not hydrated" or whatever.

Reply:
We already have this -- see `types.ensureEnumHydrated` or something like that. It is checked before accessed to any of the enum metadata.